smart_open/s3.py

```diff
@@ -199,13 +201,30 @@ def seek(self, position):
         else:
             self._body = _get(self._object, self._version_id, Range=range_string)['Body']
 
-    def read(self, size=-1):
-        if self._position >= self._content_length:
-            return b''
+    def _read_from_body(self, size=-1):
         if size == -1:
             binary = self._body.read()
         else:
             binary = self._body.read(size)
+        return binary
+
+    def read(self, size=-1):
+        if self._position >= self._content_length:
+            return b''
+
+        if self._body is None:
+            # After seek, or the first read, the self._body is None.
```

Comment:
how about closed?

Reply:
After init, the self._body is None, the connection is not built until the first read.

Reply:
So you can still read from a `closed` reading stream, interesting. Is this work as designed?

Reply:
Yes, we could read from a closed stream, even if after the `close()` operation, but there is no such api offer to users. 
In addition, according to the botocore [reference](https://botocore.amazonaws.com/v1/documentation/api/latest/reference/response.html#botocore-response), streaming body doesn't have the `closed` attribute.